### PR TITLE
Switch tokio-native-tls -> tokio-rustls

### PR DESCRIPTION
**Pull request by oldgalileo**
- Enables using PEM with certificates and private key side-by-side
- Resolves problem unique to macOS's SecureTransport resulting in users seeing:
    `a TLS error occurred: Unknown format in import.`

@tarkah I've confirmed that this does fix the issue on macOS

Fixed feature regressions:
- [x] Disabling certificate validation no longer works (although in the original it was always disabled 👻 )
- [x] Adding a specific root cert no longer works
- [x] Splitting the certificate and the private key no longer works

### CODE REVIEW

File: data/src/server.rs [hunk 1]
```diff
@@ -77,6 +77,7 @@ impl Map {
                         *password = Some(pass);
                     },
                     Sasl::Plain { password: Some(_), password_file: None, .. } => {},
+                    Sasl::External { .. } => {},
```
Comment (casperstorm) [inline data/src/server.rs:80]: I think we can turn the entire `match sasl` into a `if let`:

```rust
if let Sasl::Plain { password: password @ None, password_file: Some(pass_file), .. } = sasl {
    let pass = fs::read_to_string(pass_file)?;
    *password = Some(pass);
}
```

Comment (tarkah) [reply data/src/server.rs]: resolved by f47da01

**Review by tarkah — APPROVED** [review]
Huge thanks @oldgalileo!

File: irc/src/connection/tls.rs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+    for cert in rustls_native_certs::load_native_certs()? {
+        roots.add(cert).unwrap();
+    }
+    if let Some(cert_path) = root_cert_path {
+        let cert_bytes = fs::read(&cert_path).await?;
+        let certs =
+            rustls_pemfile::certs(&mut Cursor::new(&cert_bytes)).collect::<Result<Vec<_>, _>>()?;
+        roots.add_parsable_certificates(certs);
+    }
```
Comment (oldgalileo) [inline irc/src/connection/tls.rs:33]: This is only relevant if `accept_invalid_certs == false`

Comment (tarkah) [reply irc/src/connection/tls.rs]: Good point 364db2e

**Pull request merged**